tests/Backend/Snowflake/TypedTableWorkspacesLoadTest.php

```diff
@@ -0,0 +1,811 @@
+    {
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+        $workspace = $this->initTestWorkspace();
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $tableId = $this->createTableLanguagesMoreColumns(['Id']);
+
+        // first load
+        $options = [
+            'input' => [
+                [
+                    'source' => $tableId,
+                    'destination' => 'languagesDetails',
+                    'whereColumn' => 'iso',
+                    'whereValues' => ['dd', 'xx'],
+                    'columns' => [
+                        [
+                            'source' => 'Id',
+                            'type' => 'integer',
+                        ],
+                        [
+                            'source' => 'Name',
+                            'type' => 'varchar',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $options = InputMappingConverter::convertInputColumnsTypesForBackend(
+            $workspace['connection']['backend'],
+            $options,
+        );
+
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
+
+        // second load
+        $options = [
+            'input' => [
+                [
+                    'incremental' => true,
+                    'source' => $tableId,
+                    'destination' => 'languagesDetails',
+                    'whereColumn' => 'iso',
+                    'whereValues' => ['ff', 'xx'],
+                    'columns' => [
+                        [
+                            'source' => 'Id',
+                            'type' => 'integer',
+                        ],
+                        [
+                            'source' => 'Name',
+                            'type' => 'varchar',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $options = InputMappingConverter::convertInputColumnsTypesForBackend(
+            $workspace['connection']['backend'],
+            $options,
+        );
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
+        $this->assertEquals(5, $backend->countRows('languagesDetails'));
+    }
+
+    public function testSecondsFilter(): void
+    {
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+        $workspace = $this->initTestWorkspace();
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $tableId = $this->createTableLanguages();
+        $importFile = __DIR__ . '/../../_data/languages.csv';
+        $importCsv = new CsvFile($importFile);
+        $originalFileLinesCount = count(iterator_to_array($importCsv));
+        sleep(35);
+        $startTime = time();
+        $this->_client->writeTableAsync($tableId, $importCsv, [
+            'incremental' => true,
+        ]);
+        $this->_client->writeTableAsync($tableId, $importCsv, [
```

Comment:
proc ten load potrebujeme dvakrat? abychom si byli super jisti?

Reply:
netuším proč to tam Marc v roce 2016 dal 2x :D https://github.com/keboola/storage-api-php-client/pull/78